Assets/BossRoom/Scripts/Editor/SceneBootstrapper.cs

```diff
@@ -34,6 +34,8 @@ public class SceneBootstrapper
         const string k_LoadBootstrapSceneOnPlay = "Boss Room/Load Bootstrap Scene On Play";
         const string k_DoNotLoadBootstrapSceneOnPlay = "Boss Room/Don't Load Bootstrap Scene On Play";
 
+        const string k_TestRunnerObjectName = "Code-based tests runner";
```

Comment:
That seems like the kind of thing that would break super easily. In 6 months, someone will remove the "s" from "tests" and we'll be wondering why our tests are breaking all of a sudden.
Couldn't you set some static "isTesting" value in your test fixture setup?
Or detect command line args typical for testing?
Or

Reply:
So coming back to this, the reason this check exists is that TestRunner API is currently all internal. I'll point you to this [ticket](https://jira.unity3d.com/browse/DSTP-1082) I've raised with the Test Platform team: it will be available in a future release.

For now, this check should suffice IMO. Back to the conversation Phil initially raised, methods with `InitializeOnLoadMethod` attribute are invoked after the TestRunner scene is created & launched. My thought was also to set some `IsTesting` flag inside those or even on methods with [Setup] or [UnitySetUp], but those two are invoked per test, again much after the test scene has been launched.

Reply:
ah I see
how about OneTimeSetup, does that work or is that too late as well?
https://docs.unity3d.com/Packages/com.unity.test-framework@1.1/manual/reference-actions-outside-tests.html

And you can't read command line args like System.Environment.CommandLineArgs? (no need to go through test's API)

Reply:
After testing most of the attributes on that page, they're all far too late in the process to be able to catch inside of SceneBootstrapper.

Order is:
* Launching a test through the TestRunner window creates "InitTestScene" with a timestamp. Inside that scene is a test GameObject (I was using this object as the check previously).
* ExitEditMode (this is when we launch Startup scene), EnterPlayMode
* Tests are executed (so here any class/methods with attributes would be invoked).

And no, command line args unfortunately do not reveal anything about TestRunner when launched through the TestRunner window.